packages/client/src/object/fetchPage.ts

```diff
@@ -58,8 +63,8 @@ export function objectSetToSearchJsonV2(
   expectedApiName: string,
   existingWhere: SearchJsonQueryV2 | undefined = undefined,
 ): SearchJsonQueryV2 | undefined {
-  if (objectSet.type === "base") {
-    if (objectSet.objectType !== expectedApiName) {
+  if (objectSet.type === "base" || objectSet.type === "interfaceBase") {
+    if (objectSet.type === "base" && objectSet.objectType !== expectedApiName) {
```

Comment:
We now have two types of object sets that the old API could theoretically support, so have to check for either in case our users turn the feature flag on.